### ISSUE TITLE
Samples in the leaf

### ISSUE THREAD

**Issue by noycohen100** [Doctorado-ML/STree#50]
Hi,
Thanks for your package.
How can I get the samples at each leaf node?

**Comment by rmontanana** [comment]
Hi!

In the node constructor you can find this code:
```python
def __init__(
        self,
        clf: SVC,
        X: np.ndarray,
        y: np.ndarray,
        features: np.array,
        impurity: float,
        title: str,
        weight: np.ndarray = None,
        scaler: StandardScaler = None,
    ):
        self._clf = clf
        self._title = title
        self._belief = 0.0
        # Only store dataset in Testing
        self._X = X if os.environ.get("TESTING", "NS") != "NS" else None
        self._y = y
        self._down = None
        self._up = None
        self._class = None
        self._feature = None
        self._sample_weight = (
            weight if os.environ.get("TESTING", "NS") != "NS" else None
        )
        self._features = features
        self._impurity = impurity
        self._partition_column: int = -1
        self._scaler = scaler
```

That means that the samples aren't stored in the nodes unless in test time. This is due to memory issues if you work with large datasets.

But, if you want you can override it this way:
```python
import os
from sklearn.datasets import load_iris
from stree import Stree

os.environ["TESTING"] = "1"
X, y = load_iris(return_X_y=True)
clf = Stree()
clf.fit(X, y)
for node in clf:
    if node.is_leaf():
        print(f"{node._X.shape[0]} samples in node {node._title}")
```

Hope this helps

**Comment by noycohen100** [comment]
Thanks!
Is it possible to get the identifiers of all the samples in a specific leaf node?

**Comment by rmontanana** [comment]
I don't know what you mean with "identifiers of all the samples", could you rephrase it?

**Comment by noycohen100** [comment]
Yes, I need to get the samples' indices of each leaf.
Actually, I try to implement the function "apply", available for Random
Forest classifier (sklearn) for instance.
[image: image.png]


‫בתאריך יום ג׳, 23 בנוב׳ 2021 ב-14:01 מאת ‪Ricardo Montañana Gómez‬‏ <‪
***@***.***‬‏>:‬

**Comment by noycohen100** [comment]
Thank you!
